#! /usr/bin/env python

PACKAGE = "robotx_control"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("left_throttle_scale", double_t, 0, "left throttle * m/s", 1.0, 0.5, 1.5)
gen.add("right_throttle_scale", double_t, 0, "right throttle * m/s", 1.0, 0.5, 1.5)
gen.add("left_throttle_offset", double_t, 0, "left throttle + m/s", 0, -0.5, 0.5)
gen.add("right_throttle_offset", double_t, 0, "right throttle + m/s", 0, -0.5, 0.5)
gen.add("max_speed", double_t, 0, "max speed in m/s", 1.3, 0.2, 1.5)
gen.add("yaw_rate", double_t, 0, "yaw rate in s", 0.1, 0.05, 0.5)
gen.add("steering_direction", bool_t, 0, "CCW", True)
gen.add("left_throttle_polarity", bool_t, 0, "FWD", True)
gen.add("right_throttle_polarity", bool_t, 0, "FWD", True)

exit(gen.generate(PACKAGE, "robotx_control", "CalibrateMotor"))
